### PR TITLE
Stab chain testing

### PR DESCRIPTION
**Pull request by EwanGilligan**
Further testing for the stabilizer chain construction. This includes a fix for the stabilizer chains.

### CODE REVIEW

File: src/group/mod.rs [hunk 1]
```diff
@@ -109,6 +109,7 @@ impl Group {
 
     /// Creates the Klein 4 group
     pub fn klein_4() -> Self {
+        //TODO fix this as this isn't K_4
```
Comment (WizardOfMenlo) [inline src/group/mod.rs:112]: The Klein Group should be isomorphic to the dihedral group of order 4, if it isn't it's probably a bug in D_2n?

File: src/group/orbit/factored_transversal.rs [hunk 2]
```diff
@@ -14,6 +14,30 @@ pub struct FactoredTransversal {
     pub(super) transversal: HashMap<usize, Permutation>,
 }
 
+#[allow(clippy::implicit_hasher)]
```
Comment (WizardOfMenlo) [inline src/group/orbit/factored_transversal.rs:17]: Just for completeness, we can fix this lint by adding a type parameter <S: std::hash::BuildHasher>.
and changing transversal: &HashMap<usize, Permutation, S>.

File: src/group/orbit/factored_transversal.rs [hunk 3]
```diff
@@ -14,6 +14,30 @@ pub struct FactoredTransversal {
...
+pub fn representative_raw(
```
Comment (WizardOfMenlo) [inline src/group/orbit/factored_transversal.rs:18]: Probably be worth making it pub(crate)

File: src/group/mod.rs (hunk 1 above)
Comment (EwanGilligan) [reply src/group/mod.rs:112]: Yes I think it is a bug with the generation method for D_2n not working for n < 3 (I believe). I didn't fix that in this, but was going to open a pull request for that.

File: src/group/orbit/factored_transversal.rs (hunk 2 above)
Comment (EwanGilligan) [reply src/group/orbit/factored_transversal.rs:17]: Done now

File: src/group/orbit/factored_transversal.rs (hunk 3 above)
Comment (EwanGilligan) [reply src/group/orbit/factored_transversal.rs:18]: Done now

**Pull request merged**